pages/utils/job_utils.py

```diff
@@ -36,42 +27,92 @@
     font=dict(size=18, color="orange"),
 )
 
+def _loading_graph(done, queued, retry, failed):
+    colors = ['gold', 'mediumturquoise', 'lightgreen', 'black']
+
+    fig = go.Figure(
+        data=[
+            go.Pie(
+                    labels=['Done','Queued', 'Retry', 'Failed'],
+                    values=[done, queued, retry, failed]
+                )
+            ]
+        )
+    fig.update_traces(hoverinfo='label', textinfo='value', textfont_size=20,
+                    marker=dict(colors=colors, line=dict(color='#000000', width=2)))
+
+    return fig
 
 def handle_job_state(jm, func, repolist):
     """
     All visualizations use this interface
     to handle whether or not the job queue / result cache
     has the data that they need.
     """
+    num_done = 0
```

Comment:
explain the use of these

Reply:
will add it in the documentation.

Reply:
@JamesKunstle sounds good, will mark as resolved once fixed

Reply:
added!